### PR TITLE
Clean up some formatting

### PR DESCRIPTION
**Pull request by thinkyhead**
- Also partial cleanup of Marlin_main.cpp with a smidgen of Doxygen.

### CODE REVIEW

**Comment by galexander1** [comment]
I'm not really against this patch, but this sort of change is a good
example of why it's nice to permit feedback from others before merging
your own commits.

If we're going to be making style changes, we should have a style guide
somewhere, probably in Documentation/, so that it is a monotonic process
(moving steadily towards a consistent style, instead of intermittent
spurts in random directions).  Maybe I just didn't see it?

As for the style you used, my only objection is replacing 0.0 with 0 in
initializers.  0 is functionally fine, but 0.0 is a notice to anybody
reading the code that we are dealing with floating point values here.
I'm not sure I'd ever bother to change 0 to 0.0, but I would certainly
never change 0.0 to 0 (you know, unless the end result was int).

Other than that, I'm simply glad to see someone poking at the cobwebs.

**Comment by thinkyhead** [comment]
@galexander1 Sorry! A style guide is under construction, and we have a thread about it in the recent issues list. This is most definitely in the direction of the style guide, which aims to remain consistent to existing conventions while improving formatting and readability. You will see comments that start with `/**` and others that begin with `///<` – these are Doxygen comments that only apply to declarations and headings.

As for things like initializing an array of floats as `{ 0, 0, 0, 0 }` I consider that OK. It is clear from the context that it is an array of floats. The compiler produces the same code, and coders know this. The casual reader knows that this is a semantic shorthand for making an array of 4 initialized floats. So this is a stand for expressive clarity in that we don't need to be _overly pedantic_ where the compiler gets the point. When it is possible to say "5" (instead of "5." or "5.0" or "5f" or "5.f" or "5.0f") that is encouraged. The only time we need these other forms – and this is why they should stand out as conspicuous – is when you want to coerce the type in some direction. But of course, the compiler always promotes ints to floats, and only ever demotes floats to ints when they are passed as int parameters, assigned to int, or they are explicitly coerced. There is no compiler directive (that I know of) that changes this fundamental behavior.

As for potential disruption of other development, I'm doing my best to make these changes gradually, in regions of code that are less active. Some of these may be disruptive to forks, but truly, any active Github repo is a hassle for custom forks to keep up with. The best option for many forks in this busy time, as they are finding, is to start making Pull Requests for their custom additions, one small silver lining....

Check the thread(s) on the subject of code formatting, and there's a "documentation" issue as well....

**Comment by thinkyhead** [comment]
Well [the next thing I'm submitting](https://github.com/thinkyhead/Marlin/compare/break_out_gcode_handlers) (moving most GCode handlers into their own functions) is I think probably much less controversial! There are some style changes, but nothing too weird.

In the process of preparing my commit I was looking at lines like this...

```
filament_size[1] = (filament_size[1] == 0.0 ? DEFAULT_NOMINAL_FILAMENT_DIA : filament_size[1]);
```

and thinking I really do like this better:

```
filament_size[1] = filament_size[1] ? filament_size[1] : DEFAULT_NOMINAL_FILAMENT_DIA;
```

but then it's obvious that a better line would really be:

```
if (! filament_size[1]) filament_size[1] = DEFAULT_NOMINAL_FILAMENT_DIA;
```

...which is the most explicit in saying that "initialization happens if the value is Zero." The same syntax works for any numeric type, so we can focus on what it does and not so much on what it is, which is established at the top.

The compiler ends up compiling them all into the same resulting code, and I think humans probably find the more concise versions easier to understand as well. Where possible, I admittedly prefer concision, but I am pretty careful to preserve clarity. Oftentimes re-writing exposes naive assumptions and minor mistakes, reveals redundant and unreachable code-paths. I find that the more concise and high-level code can be made to appear, the less clutter is present in the visual field, and more apparent is the higher level structure. But then I dream in code, not everyone does I know....

**Comment by galexander1** [comment]
I'm going to take issue with some things but I just want to point out
that this is getting a little philosophical and while I'm always right,
there is room for differing opinions. :)

Doxygen is useful for code that implements an API, or for projects which
will be read like a book (i.e., education).  Spend your time
correcting/deleting wrong comments, not forcing comments into an
arbitrary format.

Regarding "0.0", that is for the human, not for the compiler.  It is an
artistic practice where we try to be clear, and style guides and language
specifications cannot make all the decisions for us.  Unless there is a
bug, or restructuring, I would not second guess the inconsequential
decisions of the original artist.

Regarding changes that make merging harder, I totally agree that this
should not be our first concern.  I do think we must nominate a stable
branch first, though.  A blessed starting point for vendors to make their
device-specific forks from.  Do we tag stable now?  Do we hold off on
gratuitous patches instead?

In other words, we have a challenging problem with nested #ifs and run-on
code that needs to be factored tastefully into smaller functions.  We
have another problem where the authors of many distinct features (such as
firmware retraction, bed leveling, and delta support) had their own
inconsistent ideas about how the planner operates, which has made
understanding its global state a bit of a mess.  I would not waste time
with whitespace, doxygen, or ".0".

I know that's a bit of a buzzkill, because it's intimidating to get into
the core.  I do think if we start seeing ambitious pull requests in this
direction, we do have a strong enough community that if you let it sit
for a week there will be a good discussion -- you do not need to worry
about making Big Decisions in a vacuum.

**Comment by thinkyhead** [comment]
With the editor I use (Sublime 2) whitespace matters for code folding. So indentation is something that I care about, especially as a "spearheading" developer of Marlin. Fugly code is messing with my flow. ;-) And again, I'm not doing anything against precedent.

True that Doxygen is useful for APIs, but it can help to generate a bird's eye view of any code base. I've used it a lot in the past to generate documentation, and some elements of Marlin have simple APIs. There are some other nice automated tools we can use to generate Markdown or HTML from the source code, but of course none that describe the parameters. I lament that Github doesn't have a view that converts Doxygen into code docs. In any case, when it comes to choosing some standard for comment blocks, I've discussed some in the coding standards topics, and Doxygen seems to be the most universal and simple. So, that's the format I'm advocating for.

**Comment by galexander1** [comment]
The simple and universal standard for comment blocks is that comments
begin with /*, then consist of a string of human-language characters
describing the situation, then end in */.  :P

**Comment by thinkyhead** [comment]
Sorry, I should have said "_among preprocessor-friendly source-code documentation formats_, Doxygen seems to be the most universal and simple."
